app/src/main/java/com/example/teamproject/MyLocation.java

```diff
@@ -0,0 +1,124 @@
+package com.example.teamproject;
+import android.content.Context;
+import android.location.Location;
+import android.location.LocationListener;
+import android.location.LocationManager;
+import android.os.Bundle;
+
+import java.util.Timer;
+import java.util.TimerTask;
+
+public class MyLocation extends Location{
+    Timer timer1;
+    LocationManager lm;
+    LocationResult locationResult;
+    boolean gps_enabled=false;
+    boolean network_enabled=false;
+
+//    public MyLocation {
+//        //empty constructor
+//    }
+
+
+    public MyLocation(String provider, Timer timer1, LocationManager lm, LocationResult locationResult, boolean gps_enabled, boolean network_enabled, LocationListener locationListenerGps, LocationListener locationListenerNetwork) {
+        super(provider);
+        this.timer1 = timer1;
+        this.lm = lm;
+        this.locationResult = locationResult;
+        this.gps_enabled = gps_enabled;
+        this.network_enabled = network_enabled;
+        this.locationListenerGps = locationListenerGps;
+        this.locationListenerNetwork = locationListenerNetwork;
+    }
+
+
+    public MyLocation(Location l, Timer timer1) {
+        super(l);
+        this.timer1 = timer1;
+    }
+
+    public boolean getLocation(Context context, LocationResult result)
+    {
+        //I use LocationResult callback class to pass location value from MyLocation to user code.
+        locationResult=result;
+        if(lm==null)
+            lm = (LocationManager) context.getSystemService(Context.LOCATION_SERVICE);
+
+        //exceptions will be thrown if provider is not permitted.
+        try{gps_enabled=lm.isProviderEnabled(LocationManager.GPS_PROVIDER);}catch(Exception ex){}
+        try{network_enabled=lm.isProviderEnabled(LocationManager.NETWORK_PROVIDER);}catch(Exception ex){}
```

Comment:
Comment above on swallowing all exceptions. (hint: it's bad because it hides any unhandled errors that you need to be aware of)